src/main/java/com/snowflake/kafka/connector/records/SnowflakeTableStreamingRecordMapper.java

```diff
@@ -0,0 +1,83 @@
+package com.snowflake.kafka.connector.records;
+
+import static com.snowflake.kafka.connector.Utils.TABLE_COLUMN_CONTENT;
+import static com.snowflake.kafka.connector.Utils.TABLE_COLUMN_METADATA;
+
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.core.JsonProcessingException;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.JsonNode;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.ObjectMapper;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.NumericNode;
+
+class SnowflakeTableStreamingRecordMapper implements StreamingRecordMapper {
+  private final ObjectMapper mapper;
+
+  public SnowflakeTableStreamingRecordMapper(ObjectMapper mapper) {
+    this.mapper = mapper;
+  }
+
+  @Override
+  public Map<String, Object> processSnowflakeRecord(
+      RecordService.SnowflakeTableRow row,
+      boolean schematizationEnabled,
+      boolean includeAllMetadata)
+      throws JsonProcessingException {
+    final Map<String, Object> streamingIngestRow = new HashMap<>();
+    for (JsonNode node : row.getContent().getData()) {
+      if (schematizationEnabled) {
+        streamingIngestRow.putAll(getMapFromJsonNodeForStreamingIngest(node));
+      } else {
+        streamingIngestRow.put(TABLE_COLUMN_CONTENT, mapper.writeValueAsString(node));
+      }
+    }
+    if (includeAllMetadata) {
+      streamingIngestRow.put(TABLE_COLUMN_METADATA, mapper.writeValueAsString(row.getMetadata()));
+    }
```

Comment:
not sure if there was some trick here but i moved it from the forEach loop above to here. I don't see any sense in parsing and putting it all over again for every jsonNode. This should improve overall processing speed as we do not serialise it for every json node we have